components/clarity-lsp/src/common/requests/definitions.rs

```diff
@@ -0,0 +1,701 @@
+use std::collections::HashMap;
+
+use super::helpers::span_to_range;
+
+use clarity_repl::analysis::ast_visitor::{traverse, ASTVisitor, TypedVar};
+use clarity_repl::clarity::vm::types::{QualifiedContractIdentifier, StandardPrincipalData};
+use clarity_repl::clarity::{ClarityName, SymbolicExpression};
+use lsp_types::Range;
+
+#[cfg(feature = "wasm")]
+#[allow(unused_imports)]
+use crate::utils::log;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum DefinitionLocation {
+    Internal(Range),
+    External(QualifiedContractIdentifier, ClarityName),
+}
+
+#[derive(Clone, Debug)]
+pub struct Definitions {
+    pub tokens: HashMap<(u32, u32), DefinitionLocation>,
+    global: HashMap<ClarityName, Range>,
+    local: HashMap<u64, HashMap<ClarityName, Range>>,
```

Comment:
Can you add some documentation comments to these so that it's easier to understand what they are without reading the implementation?